### DISCUSSION TITLE
Multiple Git Repos witihin one install

### DISCUSSION THREAD

**Discussion by ReessKennedy** [redimp/otterwiki#120] [Ideas]
## What
Given the success of https://github.com/redimp/otterwiki/discussions/87 that resulted in allowing for access to the remote Git repo, now I am wondering: Could I manage the 6 different git repos filled with docs I switch between within one Otterwiki? (Instead of having to spin up multiple installs on independent domains (e.g. project1.mydomain.com, project2.mydomain.com) ... 

## Why
- I have multiple teams and for organization we break up our docs into independent repos
- Would be great to unify management under one roof! ... for my use case this would also mean that that the user manager would have to allow me to set permissions over which repos a user has access to (e.g. don't want everyone to have access to the finance repo)

**Comment by redimp** [comment]
Hey @ReessKennedy, thanks for bringing this up. The short answer is: Currently way out of scope.

Since I run multiple instances of An Otter Wiki, too, I thought about this already: Unfortunately is this not possible without rewriting a lot of the codebase and make major changes to the database structure. For the moment this is out of scope. Sorry :/

**Reply by ReessKennedy** [reply]
I figured! Excellent work with this tool!

**Reply by ReessKennedy** [reply]
New Idea💡 on this: Think it would work if I hacked a personal fork that just checked the URL and routed me to a different SQLite database and git repo based on domain? e.g.  if (wiki1.domain.com)  { wiki1.db + wiki1.git } elseif (wiki2.domain.com) { wiki2.db + wiki2.git } 

This is based on the understanding that settings are just stored in a SQLite database and actual data in the git repo. 

Seems like it could be a simple hack to get this to go without altering much code but I am not a python expert and haven't pull things apart yet.

**Reply by redimp** [reply]
I think it will be hard (as in lots of work) to hack this in the current code. The flask app is instanciated with the settings read on start and the storage (with the git repo path) and the database object are instanciated in the `otterwiki.server` module.

**Reply by ReessKennedy** [reply]
> I think it will be hard (as in lots of work) to hack this in the current code. The flask app is instanciated with the settings read on start and the storage (with the git repo path) and the database object are instanciated in the `otterwiki.server` module.

Okay! Thanks for the warning on this. Always easy to be like, "I could probably bend this in this way" and then I am many hours in and cursing myself while simultaneously being very resistant to giving up the challenge.

**Comment by ReessKennedy** [comment]
Hey @redimp: Continued great work with this. With your setup do you run multiple Otters on subdomains of the same domain?

**Reply by redimp** [reply]
I run multiple otterwikis on the same top level domain, like

- otterwiki.com
- demo.otterwiki.com
- testing.otterwiki.com

But these are separated deployments, they know nothing about each other.
